vertical-pod-autoscaler/e2e/updater.go

```diff
@@ -152,48 +320,91 @@ func setupPDB(f *framework.Framework, name string, maxUnavailable int) *policyv1
 	return pdb
 }
 
-func makePodSet(pods *apiv1.PodList) map[string]bool {
-	result := make(map[string]bool)
+type podSet map[string]types.UID
+
+func makePodSet(pods *apiv1.PodList) podSet {
+	result := make(podSet)
+	if pods == nil {
+		return result
+	}
 	for _, p := range pods.Items {
-		result[p.Name] = true
+		result[p.Name] = p.UID
 	}
 	return result
 }
 
-func waitForPodSetChangedInDeployment(c clientset.Interface, deployment *appsv1.Deployment, podList *apiv1.PodList) error {
+func getCurrentPodSetForDeployment(c clientset.Interface, d *appsv1.Deployment) podSet {
+	podList, err := framework.GetPodsForDeployment(c, d)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	return makePodSet(podList)
+}
+
+func waitForPodSetChanged(c clientset.Interface, getCurrentPodList getCurrentPodListFunc,
+	podList *apiv1.PodList) error {
 	initialPodSet := makePodSet(podList)
 
 	err := wait.PollImmediate(pollInterval, pollTimeout, func() (bool, error) {
-		return isPodSetChanged(c, deployment, initialPodSet)
-
+		currentPodList, err := getCurrentPodList()
+		if err != nil {
+			return false, err
+		}
+		currentPodSet := makePodSet(currentPodList)
+		return isPodSetChanged(c, currentPodSet, initialPodSet), nil
 	})
 
 	if err != nil {
-		return fmt.Errorf("Waiting for set of pods changed in %v: %v", deployment.Name, err)
+		return fmt.Errorf("Waiting for set of pods changed: %v", err)
 	}
 	return nil
 }
 
-func isPodSetChanged(c clientset.Interface, deployment *appsv1.Deployment, initialPodSet map[string]bool) (bool, error) {
-	count, err := getPodSetChanges(c, deployment, initialPodSet)
-	if err != nil {
-		return false, err
-	}
-	return count > 0, nil
+func isPodSetChanged(c clientset.Interface, currentPodSet podSet, initialPodSet podSet) bool {
+	return getPodSetChanges(c, currentPodSet, initialPodSet) > 0
 }
 
-func getPodSetChanges(c clientset.Interface, deployment *appsv1.Deployment, initialPodSet map[string]bool) (int, error) {
-	currentPodList, err := framework.GetPodsForDeployment(c, deployment)
-	if err != nil {
-		return 0, err
-	}
-	currentPodSet := makePodSet(currentPodList)
+func getPodSetChanges(c clientset.Interface, currentPodSet podSet, initialPodSet podSet) int {
```

Comment:
Based by the function usage it might not be needed but based on the name itself I'd also expect new pods that are in currentPodSet to count towards diff (e.g. there is a new name in currentPodSet and no such name in initialPodSet).

Since that would change the expected results maybe we want a different name for the function?
Sadly the only one I see is getChangesFromInitialSet but maybe you can come up with something better.

Reply:
Changed naming WDYT?

Reply:
Thanks, getRestartedPodsCount sounds good.